src/Service/CaptureService.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Adyen\Shopware\Service;
+
+use Adyen\AdyenException;
+use Adyen\Client;
+use Adyen\Service\Modification;
+use Adyen\Shopware\Entity\Notification\NotificationEntity;
+use Adyen\Shopware\Exception\CaptureException;
+use Adyen\Shopware\Handlers\KlarnaPayLaterPaymentMethodHandler;
+use Adyen\Shopware\Service\Repository\OrderRepository;
+use Adyen\Webhook\EventCodes;
+use Psr\Log\LoggerAwareTrait;
+use Shopware\Core\Checkout\Order\Aggregate\OrderDelivery\OrderDeliveryEntity;
+use Shopware\Core\Checkout\Order\Aggregate\OrderDelivery\OrderDeliveryStates;
+use Shopware\Core\Checkout\Order\Aggregate\OrderLineItem\OrderLineItemCollection;
+use Shopware\Core\Checkout\Order\OrderEntity;
+use Shopware\Core\Framework\Context;
+
+class CaptureService
+{
+    use LoggerAwareTrait;
+
+    private OrderRepository $orderRepository;
+
+    private ConfigurationService $configurationService;
+
+    private ClientService $clientService;
+
+    public function __construct(
+        OrderRepository $orderRepository,
+        ConfigurationService $configurationService,
+        ClientService $clientService
+    ) {
+        $this->orderRepository = $orderRepository;
+        $this->configurationService = $configurationService;
+        $this->clientService = $clientService;
+    }
+
+    /**
+     * @throws CaptureException
+     */
+    public function doKlarnaCapture(NotificationEntity $notification, Context $context)
+    {
+        if (
+            $this->configurationService->isDelayedCaptureActive() &&
+            $notification->getPaymentMethod() === KlarnaPayLaterPaymentMethodHandler::getPaymentMethodCode() &&
+            $notification->getEventCode() === EventCodes::AUTHORISATION
+        ) {
+            $this->logger->info('Capture for order_number ' . $notification->getMerchantReference() . ' start.');
+            $order = $this->orderRepository->getOrderByOrderNumber(
+                $notification->getMerchantReference(),
+                $context,
+                ['transactions', 'currency', 'lineItems', 'deliveries', 'deliveries.shippingMethod']
+            );
+
+            if (is_null($order)) {
+                throw new CaptureException('Order with order_number ' . $notification->getMerchantReference() . ' not found.', 1645112772);
```

Comment:
Are these exception codes random numbers? I'm not familiar with these codes

Reply:
Yes timestamps, unique codes to find the exception quicker

Reply:
Got it. What do you think @peterojo?

Reply:
Interesting, but it does not seem very practical to me to have a hardcoded unix timestamp integer attached to each exception case. If we want to introduce error codes I think it would be better to use codes that are a bit more memorable (3 or 4 characters), and then document them in a new `.md` file. For now, I believe we can better search the logs with the exception message string instead. Or do I misunderstand the point of these codes?

Reply:
It's simply to pinpoint the location in the code. Searching for the string works aswell.